lib/galaxy/util/__init__.py

```diff
@@ -1135,14 +1139,49 @@ def commaify(amount):
         return commaify(new)
 
 
+def trailing_zeros_to_powerof10(amount):
```

Comment:
a shorter version operating on numbers directly:
```
def trailing_zeros_to_powerof10(amount, threshold=5):
    if abs(amount) < 10**threshold:
        return amount
    exp = threshold - 1
    while not amount % 10**(exp+1):
        exp += 1
    if exp < threshold:
        return amount
    return f"{amount // 10**exp}\u22C510^{exp}"
```

Reply:
or even shorter:
```
def trailing_zeros_to_powerof10(amount, threshold=5):
    if amount == 0 or amount % 10**threshold:
        return amount
    exp = threshold
    while not amount % 10**(exp+1):
        exp += 1
    return f"{amount // 10**exp}\u22C510^{exp}"
```

Reply:
`while not amount % 10**(exp+1):` would run infinitely if amount is not a number of 10. 

Note that if we would like to have short code for getting scientific notation we can just `f"{number:.0e}"`. My goal here was to only remove trailing 0s.